File: src/main/java/com/logtechub/automation/pages/PageManager.java
package com.logtechub.automation.pages;

import com.logtechub.automation.pages.common.LoginPage;
import com.logtechub.automation.pages.efms.EfmsHomePage;
import com.logtechub.automation.pages.etms.EtmsHomePage;
import com.microsoft.playwright.Page;

public class PageManager {
    private final Page page;
    private LoginPage loginPage;
    private EfmsHomePage efmsHomePage;
    private EtmsHomePage etmsHomePage;

    public PageManager(Page page) {
        this.page = page;
    }

    public LoginPage loginPage() {
        if (loginPage == null) {
            loginPage = new LoginPage(page);
        }
        return loginPage;
    }

    public EfmsHomePage efmsHomePage() {
        if (efmsHomePage == null) {
            efmsHomePage = new EfmsHomePage(page);
        }
        return efmsHomePage;
    }

    public EtmsHomePage etmsHomePage() {
        if (etmsHomePage == null) {
            etmsHomePage = new EtmsHomePage(page);
        }
        return etmsHomePage;
    }
}

File: src/automation/pages/page_manager.py
from playwright.sync_api import Page

from automation.pages.efms import EfmsHomePage
from automation.pages.etms import EtmsHomePage


class PageManager:
    def __init__(self, page: Page):
        self.page = page
        self._efms_home_page: EfmsHomePage | None = None
        self._etms_home_page: EtmsHomePage | None = None

    @property
    def efms_home_page(self) -> EfmsHomePage:
        if self._efms_home_page is None:
            self._efms_home_page = EfmsHomePage(self.page)
        return self._efms_home_page

    @property
    def etms_home_page(self) -> EtmsHomePage:
        if self._etms_home_page is None:
            self._etms_home_page = EtmsHomePage(self.page)
        return self._etms_home_page
